### PR TITLE
Rework existing multi-tenancy docs

### PR DESCRIPTION
**Pull request by akeller**
## Description

Closes https://github.com/camunda/camunda-docs/issues/2990, including updates to the concepts page.

Updates made:

* Concepts page
  * We are in the C8 docs, so we can drop the heavy references to C8 since it’s implied. Cleaned that up. 
  * Reworked the order on the admonition to be Helm Charts first (global flag, super easy!). Let's keep optimizing for the Helm Chart experience in mind with any ongoing multi-tenancy work.
  * Listed component docs more simply
* Operate config
  * Updated the admonition with a link to the concepts page 
* Tasklist config
  * Matched the yaml tables on the rest of the page
  * Moved the environment variable to the description (matching Zeebe)
  * Updated the admonition with a link to the concepts page 
* Zeebe config
  *   Gateway config - lowercased the example value

Open questions: 

* ✅ - #3007 - We explain multi-tenancy in Camunda 8 - is it different from how multi-tenancy works in C7? I’m looking to simplify some of the content but also think about our migrating users from C7 to C8.
* ✅ - #3006 - Operate describes situations where Operate is enabled or disabled, or Operate is enabled, but Identity is not. Is this necessary or realistic to add this much detail? I'm confused about the user flow or use case for having a hybrid multi-tenancy enabled/disabled setup.

## When should this change go live?

- [ ] This change is not yet live and should not be merged until {ADD_DATE} (apply `hold` label or convert to draft PR)?
- [x] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [ ] This functionality is already available but undocumented.
- [ ] This is a bug fix or security concern.

## PR Checklist

- [ ] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [ ] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Comment by koevskinikola** [comment]
@akeller regarding:
> We explain multi-tenancy in Camunda 8 - is it different from how multi-tenancy works in C7? I’m looking to simplify some of the content but also think about our migrating users from C7 to C8.

There are several differences between how multi-tenancy works in C7 and C8:
1. C8 only provides multi-tenancy through a tenant identifier. The [one engine per tenant approach](https://docs.camunda.org/manual/develop/user-guide/process-engine/multi-tenancy/#one-process-engine-per-tenant) from C7 isn't possible with C8 (i.e. the shared underlying resource is k8s, so in C8 it would be a "one C8 k8s deployment per tenant" :smile:).
2. In C7, users can deploy shared resources (processes, decisions, forms) available to all tenants. In C8, there are no shared resources (at the moment).
3. In C8, data is mapped to the `<default>` tenant identifier when MT is disabled. In C7, data is mapped to a `null` tenant identifier, i.e. by default resources are shared.
4. C7 has way more options (including specifying a `tenantId`) for binding DMNs for BusinessRuleTasks, and BPMN processes for CallActivities. C8 doesn't have these options, i.e. it only binds to the latest version of a DMN/BPMN definition available to a tenant. However, this feature disparity has a larger scope than MT, so I wouldn't mention it in the MT docs.
5. Tenant checks in C7 can be disabled to perform admin/maintenance operations. This can't be done in C8, but an admin user can be authorized to all tenants, which would result in the same thing.
6. Users in C7 that are authorized for multiple tenants may perform a lot more operations without providing a `tenantId`. The C7 engine will try to infer the correct `tenantId` as much as possible. In C8, we don't do tenant inference in the Broker, and ask users to provide the `tenantId` explicitly.

**Review by koevskinikola — COMMENTED** [review]
Since this is still a draft PR, I'm leaving just a comment as my review.

File: docs/self-managed/tasklist-deployment/tasklist-configuration.md [hunk 1]
```diff
@@ -56,12 +56,14 @@ clear data separation for different departments or clients.
...
 For those running a Self-Managed Camunda 8 environment, configuring multi-tenancy in Tasklist requires specific settings:
 
-| YAML path                              | Environment variable                  | Default value | Description                                                  |
-| -------------------------------------- | ------------------------------------- | ------------- | ------------------------------------------------------------ |
-| camunda.tasklist.multi-tenancy.enabled | CAMUNDA_TASKLIST_MULTITENANCY_ENABLED | false         | Activates the multi-tenancy feature within the Tasklist app. |
+| Name                                   | Description                                                  | Default value |
+| -------------------------------------- | ------------------------------------------------------------ | ------------- |
+| camunda.tasklist.multi-tenancy.enabled | Activates the multi-tenancy feature within the Tasklist app. | false         |
+
+Multi-tenancy can also be configured using the following environment variable - `CAMUNDA_TASKLIST_MULTITENANCY_ENABLED`.
```
[the remark this answers is not in the document]
Comment (akeller) [reply docs/self-managed/tasklist-deployment/tasklist-configuration.md:63]: Great ideas here, and probably something to revisit about the overall configuration experience for each component in the future. 

Let's adopt your approach of adding it to the Description, like Zeebe. Incoming commit!

**Comment by akeller** [comment]
> @akeller regarding:
> 
> > We explain multi-tenancy in Camunda 8 - is it different from how multi-tenancy works in C7? I’m looking to simplify some of the content but also think about our migrating users from C7 to C8.
> 
> There are several differences between how multi-tenancy works in C7 and C8:
> 
> 1. C8 only provides multi-tenancy through a tenant identifier. The [one engine per tenant approach](https://docs.camunda.org/manual/develop/user-guide/process-engine/multi-tenancy/#one-process-engine-per-tenant) from C7 isn't possible with C8 (i.e. the shared underlying resource is k8s, so in C8 it would be a "one C8 k8s deployment per tenant" 😄).
> 2. In C7, users can deploy shared resources (processes, decisions, forms) available to all tenants. In C8, there are no shared resources (at the moment).
> 3. In C8, data is mapped to the `<default>` tenant identifier when MT is disabled. In C7, data is mapped to a `null` tenant identifier, i.e. by default resources are shared.
> 4. C7 has way more options (including specifying a `tenantId`) for binding DMNs for BusinessRuleTasks, and BPMN processes for CallActivities. C8 doesn't have these options, i.e. it only binds to the latest version of a DMN/BPMN definition available to a tenant. However, this feature disparity has a larger scope than MT, so I wouldn't mention it in the MT docs.
> 5. Tenant checks in C7 can be disabled to perform admin/maintenance operations. This can't be done in C8, but an admin user can be authorized to all tenants, which would result in the same thing.
> 6. Users in C7 that are authorized for multiple tenants may perform a lot more operations without providing a `tenantId`. The C7 engine will try to infer the correct `tenantId` as much as possible. In C8, we don't do tenant inference in the Broker, and ask users to provide the `tenantId` explicitly.

@koevskinikola I will handle this in a separate issue/PR - #3007

**Comment by akeller** [comment]
@koevskinikola there are changes for Operate and Tasklist in here as well (you already saw and commented on Tasklist). Feel free to pull in reviewers from those teams if you think it's necessary. Otherwise, a review from you would be fine and, as always, we can iterate on docs.

**Review by koevskinikola — APPROVED** [review]
:+1: @akeller the PR looks pretty good right now. 

The changes for the Opeate and Tasklist docs make sense to me, as well. They don't change any technical details in the docs, but clarify and improve the existing docs. So I don't think any additional reviews are necessary.

**Review by christinaausley — APPROVED** [review]

**Pull request merged**